tests/safety/test_destructive_guards.py

```
@@ -0,0 +1,324 @@
+    "docker": ("unraid_mcp.tools.docker", "register_docker_tool", "unraid_docker"),
+    "vm": ("unraid_mcp.tools.virtualization", "register_vm_tool", "unraid_vm"),
+    "notifications": ("unraid_mcp.tools.notifications", "register_notifications_tool", "unraid_notifications"),
+    "rclone": ("unraid_mcp.tools.rclone", "register_rclone_tool", "unraid_rclone"),
+    "keys": ("unraid_mcp.tools.keys", "register_keys_tool", "unraid_keys"),
+}
+
+
+class TestConfirmationGuards:
+    """Every destructive action must reject calls without confirm=True."""
+
+    @pytest.mark.parametrize("tool_key,action,kwargs", _DESTRUCTIVE_TEST_CASES, ids=_CASE_IDS)
```

Ruff (0.15.0)

[warning] Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)